### ISSUE TITLE
Updating to GCC 14.2.0 ?

### ISSUE THREAD

**Issue by darcagn** [rust-lang/rustc_codegen_gcc#561]
I'm just curious if there was any particular reason the `rust-lang/gcc` fork is still on a GCC 14.0.1 dev version from February. Merging the `gcc-mirror:gcc/releases-14.2.0` tag into the `rust-lang:gcc/master` branch merges cleanly (I put it in a [repo here](https://github.com/dreamcast-rs/gcc-rustc)) with zero conflicts, and both the `check-jit` tests and the `./y.sh test --release` tests give the same pass/fail results.

**Comment by antoyo** [comment]
We don't sync often with GCC. Is there a particular reason why you would want that?

**Comment by darcagn** [comment]
I've ported the Rust std library to Sega Dreamcast (Hitachi SH4) operating system KallistiOS using rustc_codegen_gcc. It'd be simpler to tell users interested in using this experimental Rust support that they can build a single GCC 14.2.0 toolchain with JIT enabled rather than a GCC 14.2.0 toolchain for normal development and a separate GCC 14.0.1 toolchain with JIT for Rust stuff, not to mention later toolchains can have better compatibility building on platforms like macOS in my experience.

When I saw it cleanly merged, built, and worked fine I figured I'd let you know in case you were interested in updating to a more recent stable release version instead of the February development version, and to ask and see if there was any problem with doing so that I had not foreseen.
